src/components/Api.js

```diff
@@ -0,0 +1,193 @@
+        method: "PATCH",
+        headers: {
+          authorization: this._headers.authorization,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          avatar: newLink,
+        }),
+      })
+        .then((res) => {
+          if (res.ok) {
+            return res.json();
+          }
+          return Promise.reject(`Error: ${res.status} + ${res.message}`);
+        })
+        .then((result) => {
+          return result;
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
+//update profile info
+    async patchProfile(nameVar, bioVar) {
+      return fetch(this._baseUrl + "/users/me", {
+        method: "PATCH",
+        headers: {
+          authorization: this._headers.authorization,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          name: nameVar,
+          about: bioVar,
+        }),
+      })
+        .then((res) => {
+          if (res.ok) return res.json();
+          return Promise.reject(`Error: ${res.status}`);
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
+    //CARD ROUTES
+    // get all cards
+    async getInitialCards() {
+      return fetch(this._baseUrl + "/cards", 
+        {
+            method: "GET",
+            headers: {
+                authorization: this._headers.authorization,
+                "Content-Type": "application/json",
+            },
+        })
+        .then((res) => {
+            if (res.ok) 
+              return res.json();
+            return Promise.reject(`Error getting initial cards: ${res.status}`);
+        })
+    }
+    //create new card
+    async postCards(name, link) {
+      console.log('Sending data:', { name, link });
+        return fetch(`${this._baseUrl}/cards`, {
+            method: "POST",
+            headers: 
+            {
+                authorization: this._headers.authorization,
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({ name, link }),
+        })
+        .then((res) => {
+          if (res.ok) return res.json();
+          return Promise.reject(`Error: ${res.status}`);
+        }). then ((result) => {
+          console.log('Card created successfully:', result);
+          return result;
+        })
+        .catch((err) => {
+          console.error("POST Card Error:", err);
+        });
+    }
+    async deleteCard(cardID) {
+      return fetch(this._baseUrl + `/cards/${cardID}`, {
+        method: "DELETE",
+        headers: {
+          authorization: this._headers.authorization,
+          "Content-Type": "application/json",
+        },
+      })
+        .then((res) => {
+          if (res.ok) return res.json();
+          return Promise.reject(`Error: ${res.status}`);
+        })
+        .then((result) => {
+          console.log(result);
+        });
+    }
+    async cardLikeStatus(cardID, isLiked) {
```

Comment:
Function names should start with a **verb** in infinitive (there is such a point in the checklist). It’s common in coding to do it so that functions could `speak` for themselves about what they do.